server/src/main/java/io/deephaven/server/hierarchicaltable/HierarchicalTableViewSubscription.java

```diff
@@ -322,18 +323,20 @@ private static long buildAndSendSnapshot(
                 columns, rows, destinations);
         snapshotNanosConsumer.accept(System.nanoTime() - snapshotStartNanos);
 
+        // note that keyspace is identical to position space for HierarchicalTableView snapshots
+        final RowSet snapshotRows = RowSetFactory.flat(expandedSize);
+        final RowSet keyspaceViewportRows = rows.intersect(snapshotRows);
+
         // 4. Make and populate a BarrageMessage
         final BarrageMessage barrageMessage = new BarrageMessage();
+        // We don't populate firstSeq, or lastSeq debugging information; they are not relevant to this use case.
+
         barrageMessage.isSnapshot = true;
-        // We don't populate length, snapshotRowSet, snapshotRowSetIsReversed, or snapshotColumns; they are only set by
-        // the client.
```

Comment:
Please don't delete my comment, unless you're going to document it better on BarrageMessage.